src/monitor/node_metadata.c

```diff
@@ -521,6 +532,54 @@ ReportAutoFailoverNodeHealth(char *nodeName, int nodePort,
 	SPI_finish();
 }
 
+/*
+ * ReportAutoFailoverNodeReplicationState persists the replication properties of
+ * a node.
+ *
+ * We use SPI to automatically handle triggers, function calls, etc.
+ */
+void
+ReportAutoFailoverNodeReplicationState(char *nodeName, int nodePort,
+							int candidatePriority,
+							bool replicationQuorum)
+{
+	Oid argTypes[] = {
+		INT4OID,				 /* candidate_priority */
+		BOOLOID,				 /* repliation_quorum */
+		TEXTOID,				 /* nodename */
+		INT4OID					 /* nodeport */
+	};
+
+	Datum argValues[] = {
+		Int32GetDatum(candidatePriority),	  /* candidate_priority */
+		BoolGetDatum(replicationQuorum),	  /* replication_quorum */
+		CStringGetTextDatum(nodeName),        /* nodename */
+		Int32GetDatum(nodePort)               /* nodeport */
+	};
+	const int argCount = sizeof(argValues) / sizeof(argValues[0]);
+	int spiStatus = 0;
+
+	const char *updateQuery =
+		"UPDATE " AUTO_FAILOVER_NODE_TABLE " SET "
+		"candidate_priority = $1, replication_quorum = $2, "
+		"statechangetime = now() WHERE nodename = $3 AND nodeport = $4";
```

Comment:
Why do we update `statechangetime` here, when neither `goal state` nor `reported state` has been changed?

Reply:
I treated candidate_priority and replication_quorum changes as state changes. Fine with not setting it.